api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -269,11 +273,20 @@ private User initializeUserIfNeeded() {
         return user;
 
       case ERROR:
-        log.log(Level.SEVERE, String.format(
-            "free tier project %s failed to be created", user.getFreeTierBillingProjectName()));
-        user.setFreeTierBillingProjectStatus(status);
-        return userDao.save(user);
-
+        int retries = Optional.ofNullable(user.getBillingProjectRetries()).orElse(0);
+        if (retries <= 2) {
+          this.userService.setBillingRetryCount(retries + 1);
+          fireCloudService.removeUserFromBillingProject(user.getEmail(), user.getFreeTierBillingProjectName());
```

Comment:
Any chance this is going to fail for an errored billing project? You might want to swallow any such exception and continue...

Reply:
Its possible. I will wrap in a try/catch.